app/lib/pre_assembly/file_manifest.rb

```diff
@@ -18,8 +18,13 @@ def initialize(staging_location:, csv_filename:)
       @manifest = load_manifest # this will cache the entire file manifest csv in @manifest
     end
 
-    def exists?
-      File.exist?(csv_filename)
+    def valid?
+      # check to see if any files in any file sets have both preserve and shelve set to "no" ... this makes no sense and should be marked as invalid
+      invalid_files = manifest.dig('content', :file_sets).map do |__num, file_set|
+        file_set[:files].any? { |file| file.dig(:administrative, :sdrPreserve) == 'no' && file.dig(:administrative, :shelve) == 'no' }
+      end
+
+      false if invalid_files.any?
     end
```

Comment:
the new logic for verifying if the file manifest is valid...go through all files and see if of them have both preserve and shelve set to "no"